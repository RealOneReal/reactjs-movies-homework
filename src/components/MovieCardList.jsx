import { Grid } from "@mui/material";
import MovieCard from "./MovieCard";

const MovieCardList = (props) => {
  const { movieCards } = props;
  return (
    <Grid container>
      {movieCards.map((card) => (
        <Grid item xs={9} md={2} key={card.id}>
          <MovieCard props={card} />
        </Grid>
      ))}
    </Grid>
  );
};
export default MovieCardList;
