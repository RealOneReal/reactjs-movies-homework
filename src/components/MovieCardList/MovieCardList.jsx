import { Grid } from "@mui/material";
import MovieCard from "../MovieCard/MovieCard";

const MovieCardList = (props) => {
  const { results } = props;
  return (
    <Grid container spacing={1} justifyContent="center">
      {results.map((card) => (
        <Grid item key={card.id}>
          <MovieCard {...card} />
        </Grid>
      ))}
    </Grid>
  );
};
export default MovieCardList;
