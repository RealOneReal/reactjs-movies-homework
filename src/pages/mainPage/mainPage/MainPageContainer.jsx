import { Grid } from "@mui/material";

const MainPageContainer = () => {};
