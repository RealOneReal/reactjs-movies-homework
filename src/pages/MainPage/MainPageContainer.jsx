import MovieCardList from "../../components/MovieCardList";
import DataFromTMDB from "../../mock-data/mockMovies";
const MainPageContainer = () => {
  // todo: add mockData from MovieDB
  return <MovieCardList {...DataFromTMDB} />;
};
export default MainPageContainer;
