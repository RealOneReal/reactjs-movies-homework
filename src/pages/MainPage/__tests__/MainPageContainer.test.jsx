import renderer from "react-test-renderer";
import MainPageContainer from "../MainPageContainer";
import MovieCardList from "../../../components/MovieCardList/MovieCardList";
import DataFromTMBd from "../../../mock-data/mockMovies";

jest.mock("../../../components/MovieCardList/MovieCardList", () => {
  return "MovieCardList";
});
jest.mock("../../../mock-data/mockMovies", () => {
  return {
    films: "Titanic,Batman,Brat 2",
  };
});

describe("MainPageContainer", () => {
  it("should be render", () => {
    const tree = renderer.create(<MainPageContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
