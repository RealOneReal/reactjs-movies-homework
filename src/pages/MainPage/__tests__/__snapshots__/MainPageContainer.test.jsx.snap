// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageContainer should be render 1`] = `
<MovieCardList
  films="Titanic,Batman,Brat 2"
/>
`;
